### PR TITLE
Refactor AuthAccount, PublicAccount, and Block type to singleton

### PR DESCRIPTION
**Pull request by turbolent**
- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/sema/nominal_type.go [hunk 1]
```diff
@@ -109,3 +111,11 @@ func (t *NominalType) initializeMembers() {
 		t.members = withBuiltinMembers(t, members)
 	})
 }
+
+func (t *NominalType) isContainerType() bool {
+	return t.NestedTypes != nil
```
Comment (SupunS) [inline runtime/sema/nominal_type.go:116]: Should this be as:
```suggestion
	return t.IsContainerType
```

File: runtime/sema/authaccount_type.go [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+//
+var AuthAccountType = &NominalType{
+	Name:                 "AuthAccount",
+	QualifiedName:        "AuthAccount",
+	TypeID:               "AuthAccount",
+	IsInvalid:            false,
+	IsResource:           false,
+	Storable:             false,
+	Equatable:            false,
+	ExternallyReturnable: false,
```
Comment (SupunS) [inline runtime/sema/authaccount_type.go:38]: Probably make `IsContainerType : true` as well?

File: runtime/sema/nominal_type.go (hunk 1 above)
Comment (SupunS) [reply runtime/sema/nominal_type.go:116]: Or perhaps, we don't need `IsContainerType` at all, can simply rely on the `t.NestedTypes`. But then we might have to check both nil and emptiness.

Comment (turbolent) [reply runtime/sema/nominal_type.go:116]: Oh, right, I think I had added `IsContainerType` at first, then changed my mind mid-way and thought we could just rely on `NestedTypes != nil` and didn't remove `IsContainerType`. 

Good catch!

Comment (turbolent) [reply runtime/sema/nominal_type.go:115]: Removed the unused field in 67f13e7d1723c914c3e15a13fec63ad4e1e5d44a

File: runtime/sema/authaccount_type.go (hunk 2 above)
Comment (turbolent) [reply runtime/sema/authaccount_type.go:38]: Removed the unused field in 67f13e7d1723c914c3e15a13fec63ad4e1e5d44a

**Review by SupunS — APPROVED** [review]

**Pull request merged**